src/CppUTest/TestDownlevelApi.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+* Copyright (c) 2007, Michael Feathers, James Grenning and Bas Vodde
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in the
+*       documentation and/or other materials provided with the distribution.
+*     * Neither the name of the <organization> nor the
+*       names of its contributors may be used to endorse or promote products
+*       derived from this software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE EARLIER MENTIONED AUTHORS ``AS IS'' AND ANY
+* EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL <copyright holder> BE LIABLE FOR ANY
+* DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#define CPPUTEST_BUILD_DOWNLEVEL_API
+#include "CppUTest/TestHarness.h"
+
+#ifdef CPPUTEST_USE_LONG_LONG
+
+/*
+ * These implementations exist for the case when CPPUTEST_USE_LONG_LONG is
+ * defined in the library but not in the user code, to prevent link errors
+ * due to name mangling.  (The reverse case still causes link errors.)
+ *
+ * The implementations in the other files are used when the definitions
+ * match on both sides.
+ */
+
+void UtestShell::assertLongLongsEqual(cpputest_longlong expected, cpputest_longlong actual, const char* text, const char* fileName, int lineNumber, const TestTerminator& testTerminator)
+{
+    failWith(LongLongsEqualFailure(this, fileName, lineNumber, expected, actual, text), testTerminator);
+}
+
+void UtestShell::assertUnsignedLongLongsEqual(cpputest_ulonglong expected, cpputest_ulonglong actual, const char* text, const char* fileName, int lineNumber, const TestTerminator& testTerminator)
+{
+    failWith(UnsignedLongLongsEqualFailure(this, fileName, lineNumber, expected, actual, text), testTerminator);
+}
```

Comment:
Sigh.  Should have double-checked this on Linux first I guess.  This is working perfectly in VS, but compiling with either GCC or Clang in Linux causes link failures with `CPPUTEST_USE_LONG_LONG` defined on both sides now:

    lib/libCppUTest.a(lib_libCppUTest_a-Utest.o): In function `UtestShell::assertUnsignedLongLongsEqual(unsigned long long, unsigned long long, char const*, char const*, int, TestTerminator const&)':
cpputest_build/../src/CppUTest/Utest.cpp:446: multiple definition of `UtestShell::assertUnsignedLongLongsEqual(unsigned long long, unsigned long long, char const*, char const*, int, TestTerminator const&)'
    lib/libCppUTest.a(lib_libCppUTest_a-TestDownlevelApi.o):cpputest_build/../src/CppUTest/TestDownlevelApi.cpp:48: first defined here

I don't understand why this is happening.  One definition is for `struct cpputest_ulonglong` and the other is for `unsigned long long` so they should mangle to different identifiers and not conflict, which is how it works in VS.

So maybe this won't work after all.  Or maybe this is a VS-only fix and the packaging is the GCC/Clang fix?  Seems messy.